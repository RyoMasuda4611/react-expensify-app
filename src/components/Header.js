import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to='/'>Go Home</NavLink>
        <NavLink to='/create'>Go AddExpensePage</NavLink>
        <NavLink to='/edit'>Go EditExpensePage</NavLink>
        <NavLink to='/help'>Go HelpPage</NavLink>        
    </header>
);

export default Header;