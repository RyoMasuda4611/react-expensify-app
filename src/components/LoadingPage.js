import React from 'react';

const LoadingPage = () => (
  <div className="loader">
      <img className="loader__images" src="/images/loader.gif"/>
  </div>
);

export default LoadingPage;
