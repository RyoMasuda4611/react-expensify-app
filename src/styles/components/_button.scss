.button {
  background: $blue;
  color: white;
  border: none;
  display: inline-block; //ブロック要素として扱う
  font-size: $font-size-large;
  font-weight: 300;
  padding: $s-size;
  line-height: 1;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}