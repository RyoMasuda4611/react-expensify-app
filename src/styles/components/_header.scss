.header {
  background: $dark-blue;
}

.header__content {
    align-items: center;
    justify-content: space-between;
    display: flex;
    padding: $s-size 0;
}

.header__title {
  color: white;
  text-decoration: none; // invalidate default effection
  h1 {
    margin: 0;
  }
}
