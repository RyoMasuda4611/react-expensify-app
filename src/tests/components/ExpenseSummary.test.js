import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';

test('should render ExpenseSummary correctly with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={123}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={5} expensesTotal={123456}/>);
  expect(wrapper).toMatchSnapshot();
});