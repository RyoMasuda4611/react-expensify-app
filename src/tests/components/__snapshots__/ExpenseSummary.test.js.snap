// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.23
  </h1>
</div>
`;

exports[`should render ExpenseSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    5
     
    expenses
     totalling 
    $1,234.56
  </h1>
</div>
`;
